main.go

```diff
@@ -209,19 +153,23 @@ func mainWithExitCode() exitCode {
 		return failure("Failed to parse the included tracers: %v", err)
 	}
 
-	log.Infof("Assigned ProjectID: %d HostID: %d", args.projectID, environment.HostID())
-
-	metadataCollector := hostmetadata.NewCollector(args.collAgentAddr, environment)
+	metadataCollector := hostmetadata.NewCollector(args.collAgentAddr)
+	metadataCollector.AddCustomData("os.type", "linux")
 
-	// TODO: Maybe abort execution if (some) metadata can not be collected
-	hostMetadataMap = metadataCollector.GetHostMetadata()
-
-	if bpfJITEnabled, found := hostMetadataMap["host.sysctl.net.core.bpf_jit_enable"]; found {
-		if bpfJITEnabled == "0" {
-			log.Warnf("The BPF JIT is disabled (net.core.bpf_jit_enable = 0). " +
-				"Enable it to reduce CPU overhead.")
-		}
+	hostname, _ := os.Hostname()
+	metadataCollector.AddCustomData("host.name", hostname)
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		return failure("Failed to get Linux kernel version: %v", err)
 	}
+	// OTel semantic introduced in https://github.com/open-telemetry/semantic-conventions/issues/66
+	metadataCollector.AddCustomData("os.kernel.release", kernelVersion)
+	sourceIP, err := getSourceIPAddress(args.collAgentAddr)
```

Comment:
Same here, we're not entering the root namespace when retrieving the source IP.

Reply:
Added functionality back to fetch this information from root namespace with https://github.com/elastic/otel-profiling-agent/pull/90/commits/21eeedc4c471da0f56c095d7ef46ce4bbbc7301c.